feat(specs): Dencun P2P & Engine API changes (#8852)
ethereum-optimism/optimism: specs/rollup-node-p2p.md

> Post-Canyon/Shanghai blocks are broadcast on `/optimism/<chainId>/1/blocks`.

Reviewer: Style question: do we want to change this to `Post-Canyon/Shanghai, but Pre-Ecotone`? Another "clarity versus over-verbosity" balancing act I'm wondering about.

Author: I think that's a good call, let me update these topics to list which hardforks broadcast on them

- `[REJECT]` if the `payload.timestamp` is more than 5 seconds into the future
- `[REJECT]` if the `block_hash` in the `payload` is not valid
- `[REJECT]` if the block is on the V1 topic and has withdrawals
- `[REJECT]` if the block is on a topic >= V2 and does not have withdrawals

> - `[REJECT]` if the block is on a topic >= V2 and has a non-zero amount of withdrawals

Reviewer 2: ```suggestion
- `[REJECT]` if the block is on a topic <= V2 and has a withdrawals list
- `[REJECT]` if the block is on a topic >= V2 and does not have an empty withdrawals list
```

Author: For this one I believe:
> - `[REJECT]` if the block is on a topic <= V2 and has a withdrawals list

Should be:
> - `[REJECT]` if the block is on a topic < V2 and has a withdrawals list

As the V2 topic has withdrawals. I'll update to this.

Author: Done in https://github.com/ethereum-optimism/optimism/pull/8852/commits/8d44f673c2de435079885fb189294f7dd76e6c2d

> - `[REJECT]` if the block is on a topic <= V2 and has a blob gas value set

Reviewer 2: ```suggestion
- `[REJECT]` if the block is on a topic <= V2 and has a blob gas-used value set
```

- `[REJECT]` if the block is on a topic <= V2 and has an excess blob gas value set

> - `[REJECT]` if the block is on a topic >= V3 and has a blob gas used value that is not zero

Reviewer 2: ```suggestion
- `[REJECT]` if the block is on a topic >= V3 and has a blob gas-used value that is not zero
```

- `[REJECT]` if the block is on a topic >= V3 and has an excess blob gas value that is not zero

> - `[REJECT]` if the block is on a topic >= V3 and the parent beacon block root is nil

Reviewer 2: ```suggestion
- `[REJECT]` if the block is on a topic <= V2 and the parent beacon block root is not nil
- `[REJECT]` if the block is on a topic >= V3 and the parent beacon block root is nil
```
